File: Chapter9/Python/Optional Arguments in Functions in Python/Cornwell.py
# Cornwall.py - This program computes hotel guest rates.
# Input:  Days in stay and meals included
# Output:  Hotel guest rate

# Write computeRate function here

def computeRate(days,code='0'):
    if code=='A':
        return 169*days
    elif code=='C':
        return 112*days
    else:
        return days*99.99
if __name__ == '__main__':
    rate = 0.00
    dayString = input("How many days do you plan to stay?: ")
    days = int(dayString)
    question = input("Do you want a meal plan? Y or N: ")
    if question == 'Y':
        code=input('Enter A for 3 meal per day plan, or C for only breakfast plan: ')
        rate=computeRate(days,code)
    else:
        rate=computeRate(days)
    print('The rate for your stay is: ${:.2f}'.format(rate))

    # Figure out which arguments to pass to the computeRate() function and 
    # then call the computeRate() function

File: Chapter9/Java/Overloading Methods in Java/Cornwell.java
import java.util.Scanner;

public class Cornwall {
    public static void main(String[] args) {
        Scanner s = new Scanner(System.in);
        int days;
        String dayString;
        String mealPlan;
        String question;
        double rate = 0.00;

        System.out.println("How many days do you plan to stay? ");

        dayString = s.nextLine();
        days = Integer.parseInt(dayString);
        System.out.println(" Do you want a meal plan? Y or N: ");
        question = s.nextLine();
        // Figure out which arguments to pass to the computeRate() method and
        // then call the computeRate() method
        if (question.equals("Y")) {
            System.out.println("Enter mealplan (A or C) ");
            mealPlan = s.nextLine();
            rate = computeRate(days, mealPlan);
        } else {
            rate = computeRate(days);
        }

        System.out.println("The rate for your stay is: " + rate);
        System.exit(0);

    } // End of main() method.

    public static double computeRate(int days) {
        return (days * 99.99);
    }

    public static double computeRate(int days, String couponCode) {
        if (couponCode.equals("A")) {
            return (days * 169);
        } else {
            return (days * 112);
        }
    }

    // Write computeRate methods here.

} // End of Cornwall class.
